frappe/integrations/doctype/ldap_settings/ldap_settings.py

```diff
@@ -5,136 +5,162 @@
+
 class LDAPSettings(Document):
-	def validate(self):
-		if not self.flags.ignore_mandatory:
-			self.validate_ldap_credentails()
-
-	def validate_ldap_credentails(self):
-		try:
-			import ldap
-			conn = ldap.initialize(self.ldap_server_url)
-			try:
-				if self.ssl_tls_mode == 'StartTLS':
-					conn.set_option(ldap.OPT_X_TLS_DEMAND, True)
-					if self.require_trusted_certificate == 'Yes':
-						conn.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_DEMAND)
-					conn.start_tls_s()
-			except:
-				frappe.throw(_("StartTLS is not supported"))
-
-			conn.simple_bind_s(self.base_dn, self.get_password(raise_exception=False))
-		except ImportError:
-			msg = """
-				<div>
-					{{_("Seems ldap is not installed on system.<br>Guidelines to install ldap dependancies and python package")}},
-					<a href="https://discuss.erpnext.com/t/frappe-v-7-1-beta-ldap-dependancies/15841" target="_blank">{{_("Click here")}}</a>,
-				</div>
-			"""
-			frappe.throw(msg, title=_("LDAP Not Installed"))
-
-		except ldap.LDAPError:
-			conn.unbind_s()
-			frappe.throw(_("Incorrect UserId or Password"))
-
-def get_ldap_settings():
-	try:
-		settings = frappe.get_doc("LDAP Settings")
-
-		settings.update({
-			"method": "frappe.integrations.doctype.ldap_settings.ldap_settings.login"
-		})
-		return settings
-	except Exception:
-		# this will return blank settings
-		return frappe._dict()
+    def validate(self):
+        if not self.flags.ignore_mandatory:
+            if self.ldap_search_string.endswith("={0}"):
+                if self.enabled:
+                    connect_to_ldap(server_url=self.ldap_server_url,
+                                    base_dn=self.base_dn,
+                                    password=self.get_password(raise_exception=False),
+                                    ssl_tls_mode=self.ssl_tls_mode,
+                                    trusted_cert=self.require_trusted_certificate,
+                                    private_key_file=self.local_private_key_file,
+                                    server_cert_file=self.local_server_certificate_file,
+                                    ca_certs_file=self.local_ca_certs_file)
+            else:
+                frappe.throw(_("LDAP Search String needs to end with a placeholder, eg sAMAccountName={0}"))
+
+
+def get_ldap_client_settings():
+    # return the settings to be used on the client side.
+    result = {
+        "enabled": False
+    }
+    settings = frappe.get_doc("LDAP Settings")
+
+    if settings and settings.enabled:
+        result["enabled"] = True
+        result["method"] = "frappe.integrations.doctype.ldap_settings.ldap_settings.login"
+    return result
+
+
+def connect_to_ldap(server_url,
+                    base_dn,
+                    password,
+                    ssl_tls_mode,
+                    trusted_cert,
+                    private_key_file,
+                    server_cert_file,
+                    ca_certs_file):
+    try:
+        import ldap3
```

Comment:
Hi @saurabh6790  - It wasn't added to requirements.txt since that would install it for everyone.  The way the previous ldap dependancy was set, was that you had to manually install it via pip to use it.  So what do you think, keep it as it was previously, or make it a dependancy for all installs?